### ISSUE TITLE
Microsoft Windows as WFD_Sink - fail

### ISSUE THREAD

**Issue by OliveiraICTS** [albfan/miraclecast#471]
Hi,
I´ve added a file .miraclecast at root directory with extentions example existing on /res/sinkctl.protocol-extension.example.
I used Log Level 7 during this procedores with log file in PDF format is attached here.

Seen the log, it was possible to see the RTSP sequences M1 to M7 working propperly and M16(Keepalive). By the way after reach M7 Miraclecast detected a fault and disconnected from WFD_Source - Microsoft windows. 

Can you see wat was going on?
```
 * LOGS DO MIRACLECAST - Linux Terminal Commands as root: **********************
 * LOGS DO MIRACLECAST ANTES DE INICIAR O WFD_Sink ************************ 
 INFO: miracle-wifid - revision 1.0 Dec 31 2022 14:17:04
 DEBUG: friendly-name from local hostname: devel (manager_read_name() in wifi/wifid.c:395)
 DEBUG: link: new link: wlp2s0 (3) (link_new() in wifi/wifid-link.c:71)
 DEBUG: supplicant: new supplicant for wlp2s0 (supplicant_new() in wifi/wifid-supplicant.c:2123)
 INFO: link: add link: wlp2s0
 INFO: link: manage link wlp2s0
 DEBUG: supplicant: start supplicant of wlp2s0 (supplicant_start() in wifi/wifid-supplicant.c:2661)
 DEBUG: supplicant: spawn supplicant of wlp2s0 (supplicant_spawn() in wifi/wifidsupplicant.c:2507)
 INFO: supplicant: wpa_supplicant found: /usr/sbin/wpa_supplicant
 INFO: supplicant: wpas spawned as pid:4642
 DEBUG: supplicant: open supplicant of wlp2s0 (supplicant_open() in wifi/wifid-supplicant.c:2289)
 DEBUG: supplicant: local p2p-address is: 98:54:1b:64:a3:3f (supplicant_status_fn() in wifi/wifidsupplicant.c:1843)
 * LOGS DO MIRACLECAST DURANTE A INICIALIZACAO O WFD_Sink *********************
 [1046] sinkctl.c main i[00] wfd_video_formats=40 00 01 10 0001bdeb 051557ff 00000fff 10 0000
001f 11 0780 0438, 02 10 0001bdeb 155557ff 00000fff 10 0000 001f 11 0780 0438
 [1046] sinkctl.c main i[01] wfd_audio_codecs=LPCM 00000003 00, AAC 0000000f 00, AC3
00000007 00
 [1046] sinkctl.c main i[02] wfd_display_edid=0001
00ffffffffffff0051f38f50010000000e100104a51d10ff2f0000a057499b2610484f00000001010101010
1010101010101010101011a36809c70381f403020350025a510000018000000fc00496e7465726e61
6c204c43440a000000fd003c3c9a9a0e000000000000000000000000000000000000000000000000
00000030
 [1046] sinkctl.c main i[03] wfd_connector_type=05
 [1046] sinkctl.c main i[04] microsoft_cursor=none
 [1046] sinkctl.c main i[05] microsoft_rtcp_capability=none
 [1046] sinkctl.c main i[06] wfd_idr_request_capability=1
 [1046] sinkctl.c main i[07] microsoft_latency_management_capability=none
 [1046] sinkctl.c main i[08] microsoft_format_change_capability=none
 [1046] sinkctl.c main i[09] microsoft_diagnostics_capability=none
 [1046] sinkctl.c main i[10] intel_friendly_name=miraclecast
 [1046] sinkctl.c main i[11] intel_sink_manufacturer_name=GNU Linux
 [1046] sinkctl.c main i[12] intel_sink_model_name=Arch linux
 [1046] sinkctl.c main i[13] intel_sink_device_URL=http://github.com/albfan/miraclecast
# killall -9 miracle-wifid
# killall -9 miracle-sinkctl
# systemctl stop NetworkManager.service
# systemctl stop wpa_supplicant.service
# miracle-wifid --log-level 7 &
# miracle-sinkctl --log-level 7
 [1046] sinkctl.c main i[14] wfd_uibc_capability=input_category_list=GENERIC,
HIDC;generic_cap_list=Keyboard;hidc_cap_list=Keyboard/USB, Mouse/USB, MultiTouch/USB,
Gesture/USB, RemoteControl/USB;port=none
 [ADD] Link: 3
* LOGS DO MIRACLECAST Console do WFD_Sink(Miraclecast) Command:
********************
 * LOGS DO MIRACLECAST APOS INICIALIZAR O WFD_Sink (Esperando por descoberta)
******************** 
 DEBUG: supplicant: send 'WFD_SUBELEM_SET 0 000600111c4400c8' to wpas on wlp2s0
(supplicant_set_wfd_subelements() in wifi/wifid-supplicant.c:2010)
 DEBUG: supplicant: sent P2P_FIND to wpas on wlp2s0 (supplicant_p2p_start_scan() in wifi/wifidsupplicant.c:2056)
 now running on link 3
 DEBUG: supplicant: p2p-scanning now active on wlp2s0 (supplicant_p2p_find_fn() in wifi/wifidsupplicant.c:1937)
 DEBUG: peer: new peer: ae:5a:f0:05:fa:02 @ wlp2s0 (peer_new() in wifi/wifid-peer.c:52)
 INFO: peer: add peer: ae:5a:f0:05:fa:02
 DEBUG: peer: peer ae:5a:f0:05:fa:02 @ wlp2s0 started (peer_supplicant_started() in wifi/wifidpeer.c:162)
 DEBUG: supplicant: requesting data for new peer ae:5a:f0:05:fa:02
(supplicant_event_p2p_device_found() in wifi/wifid-supplicant.c:983)
 [ADD] Peer: ae:5a:f0:05:fa:02@3
 DEBUG: peer: new peer: e0:51:63:9f:b5:76 @ wlp2s0 (peer_new() in wifi/wifid-peer.c:52)
 INFO: peer: add peer: e0:51:63:9f:b5:76
 DEBUG: peer: peer e0:51:63:9f:b5:76 @ wlp2s0 started (peer_supplicant_started() in wifi/wifidpeer.c:162)
 DEBUG: supplicant: requesting data for new peer e0:51:63:9f:b5:76
(supplicant_event_p2p_device_found() in wifi/wifid-supplicant.c:983)
 [ADD] Peer: e0:51:63:9f:b5:76@3
 DEBUG: peer: new peer: 06:e3:b0:10:d6:11 @ wlp2s0 (peer_new() in wifi/wifid-peer.c:52)
 INFO: peer: add peer: 06:e3:b0:10:d6:11
 DEBUG: peer: peer 06:e3:b0:10:d6:11 @ wlp2s0 started (peer_supplicant_started() in wifi/wifidpeer.c:162)
 DEBUG: supplicant: requesting data for new peer 06:e3:b0:10:d6:11
(supplicant_event_p2p_device_found() in wifi/wifid-supplicant.c:983)
 [ADD] Peer: 06:e3:b0:10:d6:11@3
 [PROV] Peer: 06:e3:b0:10:d6:11@3 Type: pbc PIN: 
 DEBUG: supplicant: GO Negotiation Request from 06:e3:b0:10:d6:11
(supplicant_event_p2p_go_neg_request() in wifi/wifid-supplicant.c:1086)
 [GO NEG] Peer: 06:e3:b0:10:d6:11@3 Type: pbc PIN: 
 DEBUG: supplicant: connect to 06:e3:b0:10:d6:11 via pbc/(null) (supplicant_peer_connect() in
wifi/wifid-supplicant.c:760)
 DEBUG: supplicant: p2p-scanning stopped on wlp2s0 (supplicant_event_p2p_find_stopped() in
wifi/wifid-supplicant.c:925)
 DEBUG: supplicant: set STA-MAC for 06:e3:b0:10:d6:11 from to 0a:e3:b0:10:d6:11 (via GO-NEGSUCCESS) (supplicant_event_p2p_go_neg_success() in wifi/wifid-supplicant.c:1215)
$ run 3
 * LOGS DO MIRACLECAST APOS WFD_Source (Conectar) com
WFD_Sink(Miraclecast)******************
 DEBUG: supplicant: ssid: DIRECT-SeCTS-DF-DES-0576POQv
(supplicant_event_p2p_group_started() in wifi/wifid-supplicant.c:1241)
 DEBUG: supplicant: new group: p2p-wlp2s0-0 (supplicant_group_new() in wifi/wifidsupplicant.c:510)
 DEBUG: supplicant: start remote group on new group p2p-wlp2s0-0 as client/0
(supplicant_event_p2p_group_started() in wifi/wifid-supplicant.c:1275)
 DEBUG: supplicant: dhcp-comm-p2p-wlp2s0-0: L:192.168.137.106 (supplicant_group_comm_fn()
in wifi/wifid-supplicant.c:259)
 DEBUG: supplicant: dhcp-comm-p2p-wlp2s0-0: S:255.255.255.0 (supplicant_group_comm_fn() in
wifi/wifid-supplicant.c:259)
 DEBUG: supplicant: dhcp-comm-p2p-wlp2s0-0: G:192.168.137.1 (supplicant_group_comm_fn() in
wifi/wifid-supplicant.c:259)
 [CONNECT] Peer: 06:e3:b0:10:d6:11@3
 now running on peer 06:e3:b0:10:d6:11@3
 DEBUG: connection established (sink_connected() in ctl/ctl-sink.c:424)
 NOTICE: SINK connected
 M1 *** RTSP_Resuest:WFD_Source(Miraclecast) > WFD_Sink(Windows)
 DEBUG: INCOMING:
 (sink_handle() in ctl/ctl-sink.c:362)
 M1 *** RTSP_Response:WFD_Source(Miraclecast) < WFD_Sink(Windows)
 DEBUG: OUTGOING:
 (sink_handle_options() in ctl/ctl-sink.c:49)
 M2 *** RTSP_Request:WFD_Sink(Windows) > WFD_Source(Miraclecast)
 DEBUG: OUTGOING:
OPTIONS * RTSP/1.0
Require: org.wfa.wfd1.0
CSeq: 1
RTSP/1.0 200 OK
Public: "org.wfa.wfd1.0, GET_PARAMETER, SET_PARAMETER"
CSeq: 1
 (sink_handle_options() in ctl/ctl-sink.c:72)
 M2 *** RTSP_Response:WFD_Sink(Windows) < WFD_Source(Miraclecast)
 DEBUG: INCOMING:
 (sink_req_fn() in ctl/ctl-sink.c:28)
 M3 *** RTSP_Resuest:WFD_Source(Miraclecast) > WFD_Sink(Windows)
 DEBUG: INCOMING:
OPTIONS * RTSP/1.0
Require: org.wfa.wfd1.0
CSeq: 1
RTSP/1.0 200 OK
Server: MSMiracastSource/10.00.22621.0963 guid/655C937E-1B7F-0001-B58C94657F1BD901
Public: org.wfa.wfd1.0, SETUP, TEARDOWN, PLAY, PAUSE, GET_PARAMETER,
SET_PARAMETER
CSeq: 1
GET_PARAMETER rtsp://localhost/wfd1.0 RTSP/1.0
Content-Length: 673
Content-Type: text/parameters
CSeq: 2
wfd_video_formats
wfd_audio_codecs
wfd_client_rtp_ports
wfd_display_edid
wfd_connector_type
wfd_uibc_capability
wfd2_rotation_capability
wfd2_video_formats
wfd2_audio_codecs
wfd2_video_stream_control
wfd_content_protection
wfd_idr_request_capability
intel_friendly_name
intel_sink_manufacturer_name
intel_sink_model_name
intel_sink_version
intel_sink_device_URL
microsoft_latency_management_capability
microsoft_format_change_capability
microsoft_diagnostics_capability
microsoft_cursor
microsoft_rtcp_capability
 (sink_handle() in ctl/ctl-sink.c:362)
M3 *** RTSP_Response:WFD_Source(Miraclecast) < WFD_Sink(Windows) IMPORTATE
 DEBUG: OUTGOING:
 sink_handle_get_parameter() in ctl/ctl-sink.c:154)
 M4 *** RTSP_Resuest:WFD_Source(Miraclecast) > WFD_Sink(Windows)
 DEBUG: INCOMING:
microsoft_video_formats
microsoft_max_bitrate
microsoft_multiscreen_projection
microsoft_audio_mute
microsoft_color_space_conversion
RTSP/1.0 200 OK
Content-Length: 998
Content-Type: text/parameters
CSeq: 2
wfd_content_protection: none
wfd_video_formats: 40 00 01 10 0001bdeb 051557ff 00000fff 10 0000 001f 11 0780
0438, 02 10 0001bdeb 155557ff 00000fff 10 0000 001f 11 0780 0438
wfd_audio_codecs: LPCM 00000003 00, AAC 0000000f 00, AC3 00000007 00
wfd_client_rtp_ports: RTP/AVP/UDP;unicast 7236 0 mode=play
intel_sink_device_URL: http://github.com/albfan/miraclecast
intel_sink_manufacturer_name: GNU Linux
intel_friendly_name: miraclecast
microsoft_format_change_capability: none
wfd_idr_request_capability: 1
microsoft_cursor: none
wfd_connector_type: 05
microsoft_latency_management_capability: none
microsoft_diagnostics_capability: none
wfd_display_edid: 0001
00ffffffffffff0051f38f50010000000e100104a51d10ff2f0000a057499b2610484f0000000101
01010101010101010101010101011a36809c70381f403020350025a510000018000000fc00496e74
65726e616c204c43440a000000fd003c3c9a9a0e0000000000000000000000000000000000000000
0000000000000030
microsoft_rtcp_capability: none
intel_sink_model_name: Arch linux
 (sink_handle() in ctl/ctl-sink.c:362)
 M4 *** RTSP_Response:WFD_Source(Miraclecast) < WFD_Sink(Windows)
 DEBUG: OUTGOING:
 (sink_handle_set_parameter() in ctl/ctl-sink.c:249)
 DEBUG: Got URL: rtsp://192.168.137.1/wfd1.0/streamid=0
 (sink_handle_set_parameter() in ctl/ctl-sink.c:268)
 SINK set resolution 1920x1080
 M5 *** RTSP_Request:WFD_Source(Miraclecast) > WFD_Sink(Windows) IMPORTANTE
 DEBUG: INCOMING:
 (sink_handle() in ctl/ctl-sink.c:362)
 M5 *** RTSP_Response:WFD_Source(Miraclecast) < WFD_Sink(Windows)
 DEBUG: OUTGOING:
 (sink_handle_set_parameter() in ctl/ctl-sink.c:249)
SET_PARAMETER rtsp://localhost/wfd1.0 RTSP/1.0
Content-Length: 278
Content-Type: text/parameters
CSeq: 3
wfd_video_formats: 00 00 02 10 00000100 00000000 00000000 00 0000 0000 00 none
none
wfd_audio_codecs: AAC 00000001 00
wfd_presentation_URL: rtsp://192.168.137.1/wfd1.0/streamid=0 none
wfd_client_rtp_ports: RTP/AVP/UDP;unicast 7236 0 mode=play
intel_overscan_comp: x=0, y=0
RTSP/1.0 200 OK
CSeq: 3
SET_PARAMETER rtsp://localhost/wfd1.0 RTSP/1.0
Content-Length: 27
Content-Type: text/parameters
CSeq: 4
wfd_trigger_method: SETUP
RTSP/1.0 200 OK
CSeq: 4
 M6 *** RTSP_Request:WFD_Sink(Windows) > WFD_Sink(Miraclecast)
 DEBUG: OUTGOING:
 (sink_handle_set_parameter() in ctl/ctl-sink.c:349)
 M6 *** RTSP_Response:WFD_Sink(Windows) < WFD_Sink(Miraclecast)
 DEBUG: INCOMING:
 (sink_setup_fn() in ctl/ctl-sink.c:173)
 M7 *** RTSP_Request:WFD_Sink(Windows) > WFD_Sink(Miraclecast)
 DEBUG: OUTGOING:
 (sink_setup_fn() in ctl/ctl-sink.c:202)
 M7 *** RTSP_Response:WFD_Sink(Windows) > WFD_Sink(Miraclecast)
 DEBUG: INCOMING:
SETUP rtsp://192.168.137.1/wfd1.0/streamid=0 RTSP/1.0
Transport: RTP/AVP/UDP;unicast;client_port=7236
CSeq: 2
RTSP/1.0 200 OK
Transport:
RTP/AVP/UDP;unicast;client_port=7236;server_port=49639;ssrc=24570224;rtcp-fbssrc=24570225
Server: MSMiracastSource/10.00.22621.0963 guid/655C937E-1B7F-0001-B58C94657F1BD901
Blocksize: 1328
Session: 1271007105
CSeq: 2
PLAY rtsp://192.168.137.1/wfd1.0/streamid=0 RTSP/1.0
Session: 1271007105
CSeq: 3
 (sink_req_fn() in ctl/ctl-sink.c:28)
 *** Apos M7 O Linux deu problemas **********
 DEBUG: supplicant: lost peer e0:51:63:9f:b5:76 (supplicant_event_p2p_device_lost() in wifi/wifidsupplicant.c:1007)
 DEBUG: peer: peer e0:51:63:9f:b5:76 @ wlp2s0 stopped (peer_supplicant_stopped() in wifi/wifidpeer.c:172)
 DEBUG: peer: free peer: e0:51:63:9f:b5:76 @ wlp2s0 (peer_free() in wifi/wifid-peer.c:90)
 INFO: peer: remove peer: e0:51:63:9f:b5:76
 DEBUG: supplicant: lost peer ae:5a:f0:05:fa:02 (supplicant_event_p2p_device_lost() in wifi/wifidsupplicant.c:1007)
 DEBUG: peer: peer ae:5a:f0:05:fa:02 @ wlp2s0 stopped (peer_supplicant_stopped() in wifi/wifidpeer.c:172)
 DEBUG: peer: free peer: ae:5a:f0:05:fa:02 @ wlp2s0 (peer_free() in wifi/wifid-peer.c:90)
 INFO: peer: remove peer: ae:5a:f0:05:fa:02
 [REMOVE] Peer: e0:51:63:9f:b5:76@3
 [REMOVE] Peer: ae:5a:f0:05:fa:02@3
 M16 *** RTSP_Request:WFD_Sink(Windows) > WFD_Sink(Miraclecast) Keep_Alive
 DEBUG: INCOMING:
 (sink_handle() in ctl/ctl-sink.c:362)
 free(): double free detected in tcache 2 Aborted
 DEBUG: supplicant: remove group p2p-wlp2s0-0 (supplicant_event_p2p_group_removed() in
wifi/wifid-supplicant.c:1312)
 DEBUG: supplicant: free group p2p-wlp2s0-0 (supplicant_group_free() in wifi/wifidsupplicant.c:179)
 DEBUG: supplicant: killing DHCP-process pid:4679.. (supplicant_group_free() in wifi/wifidsupplicant.c:202)
 DEBUG: caught SIGCHLD for 4679, reaping child (manager_signal_fn() in wifi/wifid.c:191)
 DEBUG: supplicant: lost peer 06:e3:b0:10:d6:11 (supplicant_event_p2p_device_lost() in wifi/wifidsupplicant.c:1007)
 DEBUG: peer: peer 06:e3:b0:10:d6:11 @ wlp2s0 stopped (peer_supplicant_stopped() in wifi/wifidRTSP/1.0 200 OK
Server: MSMiracastSource/10.00.22621.0963 guid/655C937E-1B7F-0001-B58C94657F1BD901
Date: Sat, 31 Dec 2022 17:29:26 GMT
Session: 1271007105
CSeq: 3
GET_PARAMETER rtsp://localhost/wfd1.0 RTSP/1.0
Session: 1271007105
CSeq: 5
peer.c:172)
 DEBUG: peer: free peer: 06:e3:b0:10:d6:11 @ wlp2s0 (peer_free() in wifi/wifid-peer.c:90)
 INFO: peer: remove peer: 06:e3:b0:10:d6:11
```

**Comment by albfan** [comment]
new logs with --log-date-time are easier to follow. Please provide as separate files

**Comment by OliveiraICTS** [comment]
Dear Alfran,
I Could Make miraclecast performs a mirroring for few time and crashed.  To make it works i chanted the .miraclecast config file adding missing Windows required parameters by moving wfd_video_formats to wfd2_video_formats because it was wrong and added new wfd_video_formats according with [MS-WFDPE].pdf Microsoft specification.  the .miracast config file mecomes like bellow:
[wifid]
log-level=debug

[sinkctl]
extends.wfd_video_formats=00 00 02 10 0001FEFF 3FFFFFFF 00000FFF 00 0000 0000 00 none none
extends.wfd_audio_codecs=LPCM 00000003 00, AAC 0000000f 00, AC3 00000007 00
extends.wfd_display_edid=0001 00ffffffffffff0051f38f50010000000e100104a51d10ff2f0000a057499b2610484f000000010101010101010101010101010101011a36809c70381f403020350025a510000018000000fc00496e7465726e616c204c43440a000000fd003c3c9a9a0e00000000000000000000000000000000000000000000000000000030
extends.wfd_connector_type=05
extends.microsoft_cursor=none
extends.microsoft_rtcp_capability=none
extends.wfd_idr_request_capability=1
extends.microsoft_latency_management_capability=none
extends.microsoft_format_change_capability=none
extends.microsoft_diagnostics_capability=none
extends.intel_friendly_name=miraclecast
extends.intel_sink_manufacturer_name=GNU Linux
extends.intel_sink_model_name=Arch linux
extends.intel_sink_device_URL=http://github.com/albfan/miraclecast
extends.wfd_uibc_capability=input_category_list=GENERIC, HIDC;generic_cap_list=Keyboard;hidc_cap_list=Keyboard/USB, Mouse/USB, MultiTouch/USB, Gesture/USB, RemoteControl/USB;port=none
extends.wfd2_rotation_capability=none
extends.wfd2_video_formats=40 00 01 10 0001bdeb 051557ff 00000fff 10 0000 001f 11 0780 0438, 02 10 0001bdeb 155557ff 00000fff 10 0000 001f 11 0780 0438
extends.wfd2_audio_codecs=LPCM 00000003 00, AAC 0000000f 00, AC3 00000007 00
extends.wfd2_video_stream_control=00
extends.intel_sink_version=product_ID=G4716-2000 hw_version=1.1.5.1345 sw_version=1.2.4.2451
extends.microsoft_video_formats=0000003fffff
extends.microsoft_max_bitrate=25000000
extends.microsoft_multiscreen_projection=none
extends.microsoft_audio_mute=none
extends.microsoft_color_space_conversion=none

**Comment by albfan** [comment]
what means crash? do you have a debug session or logs to check what's going on?

**Comment by OliveiraICTS** [comment]
Sorry,

 �

The problem was invalid pointer occurred at Line 313 of attached log.

 �

Regards,

 �

 �



Oliveira - José Carlos de
Desenvolvimento/Desenvolvedor Sênior
***@***.*** ***@***.***> 
www.grupoicts.com.br <http://www.grupoicts.com.br/> 

		
		

Cel: +55 61 99311.9226

 �

**Comment by OliveiraICTS** [comment]
[Log_Miraclecast_20230108_1540.txt](https://github.com/albfan/miraclecast/files/10424244/Log_Miraclecast_20230108_1540.txt)
This is the log you asked to attach.

**Comment by albfan** [comment]
I was able to reproduce locally:

    free(): invalid pointer

It takes a lot to start streaming so now with `--log-date-time` we can check what time is spent on

**Comment by OliveiraICTS** [comment]
Thank you,

I have one question that I know you are able to answer.

Whi is required to performs these commands before run miraclecast?
systemctl stop NetworkManager.service
systemctl stop wpa_supplicant.service

**Comment by albfan** [comment]
I remove the comments as offtopic for this issue.

And reopen as there is a reproducible problem for this scenario

**Comment by OliveiraICTS** [comment]
O Thank you for reopen it case....  This software is very good for who wants to understand Miracast running on Linux and Raspbery

**Comment by albfan** [comment]
here is the bt:

```
#8  0x00000000004088c1 in sink_handle_get_parameter (s=0x472cf0, m=0x47ef20) at /home/alberto/projects/c/miraclecast/src/ctl/ctl-sink.c:107
#9  0x0000000000409a3a in sink_handle (s=0x472cf0, m=0x47ef20) at /home/alberto/projects/c/miraclecast/src/ctl/ctl-sink.c:371
#10 0x0000000000409ab5 in sink_rtsp_fn (bus=0x477a30, m=0x47ef20, data=0x472cf0) at /home/alberto/projects/c/miraclecast/src/ctl/ctl-sink.c:386
#11 0x0000000000416c7d in rtsp_call (bus=0x477a30, m=0x47ef20) at /home/alberto/projects/c/miraclecast/src/shared/rtsp.c:2635
#12 0x0000000000417145 in rtsp_incoming_message (m=0x47ef20) at /home/alberto/projects/c/miraclecast/src/shared/rtsp.c:2763
#13 0x0000000000415fe4 in parser_submit (bus=0x477a30) at /home/alberto/projects/c/miraclecast/src/shared/rtsp.c:2157
#14 0x000000000041627e in parser_feed_char_header (bus=0x477a30, ch=13 '\r') at /home/alberto/projects/c/miraclecast/src/shared/rtsp.c:2253
#15 0x0000000000416951 in parser_feed_char (bus=0x477a30, ch=13 '\r') at /home/alberto/projects/c/miraclecast/src/shared/rtsp.c:2504
#16 0x0000000000416a6e in rtsp_parse_data (bus=0x477a30, 
    buf=0x7fffffffd1e0 "GET_PARAMETER rtsp://localhost/wfd1.0 RTSP/1.0\r\nSession: 2006058288\r\nCSeq: 5\r\n\r\n07-088C-270AAE2DD901\r\nDate: Sun, 22 Jan 2023 08:27:22 GMT\r\nSession: 2006058288\r\nCSeq: 3\r\n\r\n2627-2DAE-0007-088C-270AAE2DD"..., len=80) at /home/alberto/projects/c/miraclecast/src/shared/rtsp.c:2550
#17 0x000000000041722b in rtsp_read (bus=0x477a30) at /home/alberto/projects/c/miraclecast/src/shared/rtsp.c:2802
```

fixed on master 279303d

**Comment by OliveiraICTS** [comment]
Thanks

 �



Oliveira - José Carlos de
Desenvolvimento/Desenvolvedor Sênior
***@***.*** ***@***.***> 
www.grupoicts.com.br <http://www.grupoicts.com.br/> 

		
		

Cel: +55 61 99311.9226

 �
